packages/SideNavigation/SubMenu/SubMenu.jsx

```diff
@@ -0,0 +1,108 @@
+import React from 'react'
+import PropTypes from 'prop-types'
+
+import DecorativeIcon from '@tds/core-decorative-icon'
+import Box from '@tds/core-box'
+import FlexGrid from '@tds/core-flex-grid'
+import Text from '@tds/core-text'
+import { componentWithName } from '@tds/util-prop-types'
+
+import ColoredTextProvider from '../../../shared/components/ColoredTextProvider/ColoredTextProvider'
+
+import styles from './SubMenu.scss'
+
+/**
+ * Expandable content areas for use with the `SideNavigation`
+ *
+ * _This component can only be accessed as a name-spaced component: `SideNavigation.SubMenu`._
+ */
+const SubMenu = ({ children, label, onClick, id, isOpen, active }) => {
+  const options = {
+    subMenuLink: true,
+  }
+
+  return (
+    <div className={styles.mainDiv}>
+      <button
+        onClick={() => {
+          onClick(id)
+        }}
+        className={active ? styles.active : styles.button}
+      >
+        <Box vertical={3}>
+          <FlexGrid limitWidth={false} gutter={false}>
```

Comment:
I think the FlexGrid dependency can be removed in favour of a vanilla flexbox. The point here is to have the Caret icon on the far right. This can be achieved using a single class:

**Submenu.jsx**
```jsx
<Box vertical={3} inline between={2} dangerouslyAddClassName={styles.space}>
  <Text> ... </Text>
  <DecorativeIcon />
</Box>
```

**Submenu.modules.scss**
```scss
.space {
  justify-content: space-between;
}
```

Reply:
Flex grid removed